File: Projekt1_cpp/Projekt1_cpp/Guarana.cpp
#include "Guarana.h"

Guarana::Guarana(pair<int, int>pole, Swiat* swiat)
	:Roslina(0, 0, pole, swiat) {}

Organizm* Guarana::potomek(pair<int, int>pole) {
	return new Guarana(pole, swiat);
}

void Guarana::rysowanie() {
	cout << 'g';
}

string Guarana::rodzajKolizji(Organizm* organizm) {
	organizm->setSila(organizm->getSila() + 3);
	return "Wygrywa";
}

string Guarana::getNazwa() {
	return "guarana";
}

File: Projekt2_java/src/projekt2/rosliny/Guarana.java
package projekt2.rosliny;

import projekt2.*;

import java.awt.*;

public class Guarana extends Roslina {
    public Guarana(Pair pole, Swiat swiat) {
        super(0, 0, pole, swiat);
    }

    @Override
    public Organizm potomek(Pair pole) {
        return new Guarana(pole, swiat);
    }

    @Override
    public Color rysuj() {
        return Color.RED.darker();
    }

    @Override
    public String rodzajKolizji(Organizm organizm) {
        organizm.setSila(organizm.getSila() + 3);
        return super.rodzajKolizji(organizm);
    }

    @Override
    public String getNazwa() {
        return "guarana";
    }
}
